.azure-pipelines/util/get-pr-changed-files.yml

```diff
@@ -0,0 +1,18 @@
+steps:
+- task: PowerShell@2
+  displayName: List PR Changed Files
+  inputs:
+    targetType: inline
+    pwsh: true
+    script: |
+      $targetBranch = $(System.PullRequest.TargetBranch)
+      git fetch --no-tags origin "${targetBranch}:origin/base"
```

Comment:
how will this behave if:

- I got branch B and C both branch out from branch A
- have commit 1 added in B and commit 2 added in C
- create PR from B to C

Will it calculate diff between branch B and C?

Reply:
It's calculated by GitHub itself. When the PR is created and a pipeline is triggered, GitHub will create a new "ref" based your branch B compared to C. Let's call this as ref D.
In our pipeline, we only diff D and C.

Reply:
if the PR is from a fork repo, isn't the origin here means the fork repo? say the target branch is something like "ref/azure/main". How it get the main from Azure?

Reply:
PR trigger will send the ref to piepline. The origin here is always "https://github.com/Azure/azure-powershell". Refer to [build](https://dev.azure.com/azclitools/public/_build/results?buildId=203735&view=logs&j=ca395085-040a-526b-2ce8-bdc85f692774&t=319cc4f9-2fdf-5261-34c0-5e64584fddb0).